wordpress-mobile/WordPress-Android
# [Reader] `read/streams/discover` endpoint integration

Fixes #19572 

-----

## To Test:

1. [App] Install JP and sign in.
2. [App] Make sure `reader_discover_new_endpoint` is enabled (Open "Me" screen -> Debug settings -> Remote Features).
3. [Android Studio] Open `App Inspection` in Android Studio.
4. [Android Studio] Open "Network Inspector".
5. [App] Open "Reader" and select "Discover".
6. 🔍  [Android Studio] Verify `read/streams/discover` is shown in "Network Inspector". Pull-to-refresh should also make a call to this endpoint. The cards should be shown as expected.
7. [App] Repeat step 2 but this time disable `reader_discover_new_endpoint`.
8. [App] Open "Reader" and select "Discover".
9. 🔍  [Android Studio] Verify `read/tags/cards` is shown in "Network Inspector". Pull-to-refresh should also make a call to this endpoint. The cards should be shown as expected.

> [!IMPORTANT]  
> An extra small fix implemented in this PR is that we also verify if the first card in "Discover" feed is a recommended blogs card. If it is, we move it to be the third card (index 2). Previously we were only doing this for recommended tags card.

-----

## Regression Notes

1. Potential unintended areas of impact

    - "Discover" feed

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    --

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
